### PR TITLE
ui: Move peers to a subapplication

### PR DESCRIPTION
**Pull request by johncowen**
### Description
This PR moves the peering feature into a sub-application to take advantage of the extra features that offers.

Note there is a little more moving around/renaming to be done here going by the fact that we have decided to use the term `Peering` more over just `Peer` for code things.

All discussed offline.

### PR Checklist

* [ ] updated test coverage
* [ ] external facing docs updated
* [x] not a security concern

/cc @LevelbossMike

### CODE REVIEW

File: ui/packages/consul-peerings/app/components/consul/peer/search-bar/index.hbs [hunk 1]
```diff
@@ -81,7 +81,7 @@ as |key value|}}
     {{#each
       (get
         (require '/models/peer'
-          path='schema'
```
Comment (johncowen) [inline ui/packages/consul-peerings/app/components/consul/peer/search-bar/index.hbs:84]: Oh forgot to mention, this missed a PR somehow for the filtering thing, I was gonna PR it separate, but it slipped in here in the copy, I'm happy to leave it in here, but let me know if you want me to PR separate.

File: ui/packages/consul-ui/docs/bookmarklets.mdx [hunk 2]
```diff
@@ -12,7 +12,7 @@ Below is a list of the most commonly used functions as bookmarklets followed by
 | [Enable ACLs](javascript:Scenario('CONSUL_ACLS_ENABLE=1')) | Enable ACLs |
 | [Enable TProxy](javascript:Scenario('CONSUL_TPROXY_ENABLE=1')) | Enable TProxy |
 | [Enable Nspaces](javascript:Scenario('CONSUL_NSPACES_ENABLE=1')) | Enable Namespace Support |
-| [Enable Peers](javascript:Scenario('CONSUL_PEERING_ENABLE=1')) | Enable Peers Support |
+| [Enable Peers](javascript:Scenario('CONSUL_PEERINGS_ENABLE=1')) | Enable Peering Support |
```
Comment (LevelbossMike) [inline ui/packages/consul-ui/docs/bookmarklets.mdx:15]: do we need to update `get-environment` as well? I think this is where this cookie value is being read. And we are changing from `CONSUL_PEERING_ENABLE` to `CONSUL_PEERINGS_ENABLE` here.

**Review by LevelbossMike — COMMENTED** [review]
This feels a lot for making the peer route optional but 🤷 - can you maybe go into detail a bit more about the additional benefits this approach provides? just for documentation purposes.

File: ui/packages/consul-ui/docs/bookmarklets.mdx (hunk 2 above)
Comment (johncowen) [reply ui/packages/consul-ui/docs/bookmarklets.mdx:15]: I missed a couple 😬 I've added those and searched the codebase for references to that to triple check, thanks for the nudge!

**Review by evrowe — APPROVED** [review]

**Comment by johncowen** [comment]
a rebase has made some test weirdness here, just giving it all a once over before merge

**Pull request merged**